bundles/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/api/model/UniFiController.java

```diff
@@ -61,18 +61,30 @@
 
     private final int port;
 
+    private String csrfToken;
+
     private final String username;
 
     private final String password;
 
+    private final boolean unifiOS;
+
     private final Gson gson;
 
     public UniFiController(HttpClient httpClient, String host, int port, String username, String password) {
         this.httpClient = httpClient;
         this.host = host;
-        this.port = port;
+        // assume UniFi-OS on UDM if port is 443
+        if (port == 443) {
+            this.unifiOS = true;
+            this.port = -1;
```

Comment:
Won't this cause problems with the request?

Reply:
I dont wanted to change how the URI is built and this always specifies a port. But using a "-1" for port instructs the builder to use the default port for the scheme and will not put the 443 in a https-request.
See discussion [here](https://codereview.stackexchange.com/questions/25485/truncate-port-number-from-absolute-uri)